src/include/platform/PlatformManager.h

```diff
@@ -233,11 +247,48 @@ inline void PlatformManager::RunEventLoop()
     static_cast<ImplClass *>(this)->_RunEventLoop();
 }
 
+/**
+ * @brief
+ *  Starts the stack on its own task with an associated event queue
+ *  to dispatch and handle events posted to that task.
+ *
+ *  This is thread-safe.
+ *  This is *NOT SAFE* to call from within the CHIP event loop since it can grab the stack lock.
+ */
 inline CHIP_ERROR PlatformManager::StartEventLoopTask()
 {
     return static_cast<ImplClass *>(this)->_StartEventLoopTask();
 }
 
+/**
+ * @brief
+ *  This will cause the event loop to exit, preventing the processing of any further events in the
+ *  queue.
+ *
+ *  Additionally, this stops the CHIP task if the following criteria are met:
+ *      1. One was created earlier through a call to StartEventLoopTask
+ *      2. This call isn't being made from that task.
```

Comment:
It's a little weird to have this API be "stop task" when sometimes it does not, but I don't have a better suggestion... The docs will have to do, I guess.